### PR TITLE
Module 3 - Astreika

### PR DESCRIPTION
**Pull request by DmitryAstreiko**

### CODE REVIEW

File: src/service.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+			.then((data) => dispatch({ type: GET_COURSES, payload: data }));
+	} catch (error) {}
+};*/
+
+export const loadCourses = (dispatch) => {
+	try {
+		fetch('http://localhost:4000/courses/all')
+			.then((Response) => Response.json())
+			.then((data) => dispatch(loadCoursesAction(data.result)));
+	} catch (error) {}
```
Comment (barys-liutkouski-epam) [inline src/service.js:18]: this was not part of the task, but usually it's a good idea to at least `console.error` inside these catch blocks for easier troubleshooting

File: src/service.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+export const loadUser = (newUser) => (dispatch) => {
+	try {
+		fetch('http://localhost:4000/login', {
+			method: 'POST',
+			body: JSON.stringify(newUser),
+			headers: {
+				'Content-Type': 'application/json',
+			},
+		})
+			.then((Response) => Response.json())
```
Comment (barys-liutkouski-epam) [inline src/service.js:46]: variables and parameters are always named in `camelCase`

File: src/App.jsx [hunk 3]
```diff
@@ -1,84 +1,62 @@
...
-	function changeIsShowCreateCourse() {
-		setIsShowCreateCourse(!isShowCreateCourse);
+	function getCurrentCourse(courseId) {
+		return allCoursesItem.filter((x) => x.id === courseId);
 	}
 
-	function getAuthors() {
-		return mockedAuthorsList;
-	}
+	function GoToCourse() {
```
Comment (barys-liutkouski-epam) [inline src/App.jsx:40]: camelCase

File: src/components/Courses/Courses.jsx [hunk 4]
```diff
@@ -33,30 +41,80 @@ function Courses(props) {
...
+			if (resArray.length > 0) {
+				setCourses(resArray);
+			}
+		}
+	}
+
+	useEffect(() => {
+		//const currentToken = localStorage.getItem('courseUserToken');
+		//if (currentToken === null) {
+		if (isLogIn === false) {
```
Comment (barys-liutkouski-epam) [inline src/components/Courses/Courses.jsx:69]: this kind of logic is usually abstracted away into separate components, here's an example:

App.jsx
```
<Routes>
				<Route path='/' element={<AuthorizedGuard><Login /></AuthorizedGuard>} />
```

AuthorizedGuard
```
// useEffect(isLogIn) from here
```

Comment (barys-liutkouski-epam) [inline src/components/Courses/Courses.jsx:69]: this kind of logic is usually abstracted away into separate components, here's an example:

App.jsx
```
<Routes>
				<Route path='/' element={<AuthorizedGuard><Login /></AuthorizedGuard>} />
```

AuthorizedGuard
```
// useEffect(isLogIn) from here
```

Comment (DmitryAstreiko) [reply src/components/Courses/Courses.jsx:69]: https://pavankjadda.medium.com/authorization-guards-with-react-router-8ecb7900950a

Comment (DmitryAstreiko) [reply src/components/Courses/Courses.jsx:69]: https://blog.netcetera.com/how-to-create-guarded-routes-for-your-react-app-d2fe7c7b6122

**Pull request merged**